lib/puppet/resource/type.rb

```diff
@@ -413,6 +413,9 @@ def assign_defaults(resource, scope, resource_hash)
 
   def validate_resource_hash(resource, resource_hash)
     Puppet::Pops::Types::TypeMismatchDescriber.validate_parameters(resource.to_s, parameter_struct, resource_hash)
+  rescue Puppet::ParseError => e
```

Comment:
This is quite an ugly fix. Do we really need to do it this way? Isn't it possible to detect the special case beforehand and validate differently (or not at all) ?